### PR TITLE
feat: add new stripe methods for payment method id

### LINKED ISSUE #5498: Add new API methods for create customer, retry invoice, create subscription
When handling a payment, new URL methods will be created that handle creating a customer with a payment method, retrying an invoice with a new payment method, and creating a subscription.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-2005)
┆Issue Number: FXA-2005

### PR DESCRIPTION
**Pull request by bbangert**
Co-authored-by: Barry Chen <chenba@gmail.com>

Because:

* We are moving to using Payment Intent API and SCA.

This commit:

* Add's new API endpoints that return Stripe objects for the front-end
  to use to determine its next action.

Closes #5498

## Checklist

_Put an `x` in the boxes that apply_

- [X] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/validators.js [hunk 1]
```diff
@@ -292,6 +293,44 @@ module.exports.activeSubscriptionValidator = isA.object({
...
+          })
+          .unknown(true)
+          .required(),
+      })
+      .unknown(true)
+      .required(),
+  })
+  .unknown(true);
+
+module.exports.susbscriptionsInvoicePIExpandedValidator = isA
```
Comment (lmorchard) [inline packages/fxa-auth-server/lib/routes/validators.js:319]: Oh, missed this on first skim: `susbscriptionsInvoicePIExpandedValidator` should probably be `subscriptionsInvoicePIExpandedValidator`?

File: packages/fxa-auth-server/lib/routes/subscriptions.js [hunk 2]
```diff
@@ -1169,6 +1250,71 @@ const directRoutes = (
...
+    {
+      method: 'POST',
+      path: '/oauth/subscriptions/invoice/retry',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken',
+        },
+        response: {
+          schema: validators.susbscriptionsInvoicePIExpandedValidator,
```
Comment (lmorchard) [inline packages/fxa-auth-server/lib/routes/subscriptions.js:1306]: Although the typo's at least consistent over here :)

File: packages/fxa-auth-server/lib/routes/validators.js [hunk 3]
```diff
@@ -292,6 +293,44 @@ module.exports.activeSubscriptionValidator = isA.object({
   cancelledAt: isA.alternatives(isA.number(), isA.any().allow(null)),
 });
 
+// This is a Stripe subscription object with latest_invoice.payment_intent expanded
+module.exports.susbscriptionsSubscriptionExpandedValidator = isA
```
Comment (lmorchard) [inline packages/fxa-auth-server/lib/routes/validators.js:297]: Hmm, similar typo here but also consistent where it's used

File: packages/fxa-auth-server/test/local/payments/stripe.js [hunk 4]
```diff
@@ -249,6 +253,175 @@ describe('StripeHelper', () => {
...
+      const invoiceRetryExpected = deepCopy(invoiceRetry);
+      sandbox
+        .stub(stripeHelper.stripe.paymentMethods, 'attach')
+        .returns(attachExpected);
+      sandbox
+        .stub(stripeHelper.stripe.customers, 'update')
+        .returns(customerExpected);
+      sandbox
+        .stub(stripeHelper.stripe.invoices, 'retrieve')
+        .returns(invoiceRetryExpected);
```
Comment (chenba) [inline packages/fxa-auth-server/test/local/payments/stripe.js:299]: (My brain doesn't know it's a holiday and randomly remember this thing it saw...)

We should change these `stub`s where it `returns` to `resolves` when the stub'd function is an async function.  That way they'll catch errors if the callers did not `await`.

**Pull request merged**